lib/collection/response.js

```diff
@@ -107,6 +115,8 @@ _.inherit((
 
 _.assign(Response.prototype, /** @lends Response.prototype */ {
     update: function (options) {
+        // options.stream accepts new Buffer() as well as new Buffer().toJSON()
+        options.stream = normaliseStream(options.stream);
```

Comment:
@shamasis Doing this here over L144, so that `options.stream` has `byteLength` for `options.stream.byteLength` on L161.